lpvartypes.pas

```diff
@@ -1793,6 +1815,13 @@ function TLapeType.Eval(Op: EOperator; var Dest: TResVar; Left, Right: TResVar;
   begin
     Dest := NullResVar;
     Exit(Left);
+  end 
+  else if (op in CompoundOperators) and (lcoCOperators in Compiler.FOptions) then 
+  begin
+    if (not Left.Writeable) then
+      LapeException(lpeCannotAssign, _DocPos);
```

Comment:
I added that test there because for some reason I was able to assign to constants without it. No exception was raised for some reason. 
I guess it can be added to EvalConst, tho idk why you want it to be added there.